metaswitch/common/test/test_alarms.py

```diff
@@ -0,0 +1,302 @@
+# software and licensed under the openssl licenses.
+# "openssl licenses" means the openssl license and original ssleay license
+# under which the openssl project distributes the openssl toolkit software,
+# as those licenses appear in the file license-openssl.
+import unittest
+import mock
+import imp
+import atexit
+import threading
+import time
+import logging
+import threading
+
+_log = logging.getLogger()
+
+from metaswitch.common.alarms import (AlarmState,
+                                      BaseAlarm,
+                                      Alarm,
+                                      MultiSeverityAlarm,
+                                      CLEARED,
+                                      _AlarmManager)
+
+
+class TestAlarmManager(_AlarmManager):
+    """AlarmManager with extra test function."""
+
+    def wake_up(self):
+        """Force the alarm manager to check whether to re-sync."""
+        with self._condition:
+            self._condition.notify()
+
+    def safe_terminate(self):
+        """Terminate on another thread to avoid hanging."""
+        terminate_thread = threading.Thread(target=self.terminate)
+        terminate_thread.daemon = True
+        terminate_thread.start()
+        terminate_thread.join(5)
+
+
+class TestAlarmState(unittest.TestCase):
+    @mock.patch('metaswitch.common.alarms._sendrequest')
+    def test_issue_alarm(self, mock_sendrequest):
+        """Alarm states can be issued."""
+        alarm_state = AlarmState('TestIssuer', 1000, 6)
+        alarm_state.issue()
+        mock_sendrequest.assert_called_once_with(['issue-alarm',
+                                                  'TestIssuer',
+                                                  '1000.6'])
+
+
+class TestBaseAlarm(unittest.TestCase):
+    @mock.patch('metaswitch.common.alarms._sendrequest')
+    def test_clear_base_alarm(self, mock_sendrequest):
+        """Alarms can be cleared."""
+        base_alarm = BaseAlarm('TestIssuer', 1000)
+        base_alarm.clear()
+        mock_sendrequest.assert_called_once_with(['issue-alarm',
+                                                  'TestIssuer',
+                                                  '1000.1'])
+
+
+class TestAlarm(unittest.TestCase):
+    @mock.patch('metaswitch.common.alarms._sendrequest')
+    def test_set_alarm(self, mock_sendrequest):
+        """Alarms can be set."""
+        alarm = Alarm('TestIssuer', 1000, 3)
+        alarm.set()
+        mock_sendrequest.assert_called_once_with(['issue-alarm',
+                                                  'TestIssuer',
+                                                  '1000.3'])
+
+
+    @mock.patch('metaswitch.common.alarms._sendrequest')
+    def test_clear_alarm(self, mock_sendrequest):
+        """Alarm can be cleared after they are set."""
+        alarm = Alarm('TestIssuer', 1000, 3)
+        alarm.set()
+        alarm.clear()
+        mock_sendrequest.assert_called_with(['issue-alarm',
```

Comment:
There will, but I'm not trying to test that here - it's tested in another test case. The idea is that each test case tests just one thing.